pkg/builder/service_builder.go

```diff
@@ -74,6 +73,9 @@ func (b *Builder) BuildService(key types.NamespacedName, owner metav1.Object, op
 	if !opts.ExcludeSelectorLabels {
 		svc.Spec.Selector = opts.SelectorLabels
 	}
+	if opts.LoadBalancerClass != nil {
```

Comment:
You should also be providing this additional option when invoking `BuildService`:
- https://github.com/mariadb-operator/mariadb-operator/blob/c3079efacf197c17a6ecce82acd162a0d25489b9/pkg/builder/service_builder.go#L49